src/App.tsx

```diff
@@ -7,33 +8,85 @@
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type ClockProps = {
+  name: string;
+};
+
+type ClockState = {
+  time: string;
+  hasClock: boolean;
+  name: string
+};
+
+export class App extends React.Component<ClockProps, ClockState> {
+  timerId: number | null = null;
+
+  state = {
+    time: '',
+    hasClock: true,
+    name: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.hideClock);
+    document.addEventListener('click', this.showClock);
+
+    this.updateTime();
+    this.timerId = window.setInterval(this.updateTime, 1000);
+    window.setInterval(this.updateClockName, 3300);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.hideClock);
+    document.removeEventListener('click', this.showClock);
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+    if (this.timerId) {
+      clearInterval(this.timerId);
+    }
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  hideClock = (event: MouseEvent) => {
+    event.preventDefault();
+    this.setState({ hasClock: false });
+  };
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  showClock = () => {
+    this.setState({ hasClock: true });
+  };
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  updateTime = () => {
+    const today = new Date();
+    const time = today.toUTCString().slice(-12, -4);
 
-        {' time is '}
+    console.info(time);
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+    this.setState({ time });
+  };
+
+  updateClockName = () => {
+    const newClockName = getRandomName();
+
+    console.debug(`Renamed from ${this.state.name} to ${newClockName}`);
+
+    this.setState({ name: newClockName });
+  };
+
+  render() {
+    const { time, hasClock, name } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        <div>
+          {hasClock && (
```

Comment:
Replace the outer div with the Clock component and pass the 'hasClock' and 'name' props.